### PR TITLE
Cassandra 18635 5.0

### PR DESCRIPTION
**Pull request by bereng**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

**Comment by bereng** [comment]
Green 500 [repeats](https://app.circleci.com/pipelines/github/bereng/cassandra/1151/workflows/e9db3762-8689-44a5-ac3a-0dc7f04d805b/jobs/37875) and green [CI](https://app.circleci.com/pipelines/github/bereng/cassandra/1151/workflows/f5b492c3-1725-4333-ad61-a86f7ebb2d8c)

File: src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java [hunk 1]
```diff
@@ -283,6 +284,7 @@ public void resume()
         writeLock.lock();
         try
         {
+            logger.debug("Bereng resume\n" + ExceptionUtils.getStackTrace(new Exception()));
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java]: @bereng remove this please

File: src/java/org/apache/cassandra/tools/nodetool/UpgradeSSTable.java [hunk 2]
```diff
@@ -56,13 +59,25 @@ public void execute(NodeProbe probe)
...
-                probe.upgradeSSTables(probe.output().out, keyspace, !includeAll, maxSSTableTimestamp, jobs, tableNames);
-            }
-            catch (Exception e)
+            for (int retries = 0; retries < 5; retries++)
             {
-                throw new RuntimeException("Error occurred during enabling auto-compaction", e);
+                try
+                {
+                    if (retries > 0)
+                        Thread.sleep(500);
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/tools/nodetool/UpgradeSSTable.java:67]: use Uninterruptible sleep here maybe?

File: src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java [hunk 3]
```diff
@@ -301,6 +303,7 @@ public void pause()
         writeLock.lock();
         try
         {
+            logger.debug("Bereng paused\n" + ExceptionUtils.getStackTrace(new Exception()));
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java]: same

File: test/distributed/org/apache/cassandra/distributed/test/UpgradeSSTablesTest.java [hunk 4]
```diff
@@ -83,6 +83,7 @@ public void upgradeSSTablesInterruptsOngoingCompaction() throws Throwable
                 ColumnFamilyStore cfs = Keyspace.open(ks).getColumnFamilyStore("tbl");
                 CompactionManager.instance.submitMaximal(cfs, FBUtilities.nowInSeconds(), false, OperationType.COMPACTION);
             }).apply(KEYSPACE);
+
```
Comment (smiklosovic) [inline test/distributed/org/apache/cassandra/distributed/test/UpgradeSSTablesTest.java]: if we revert this then this patch is just a one-filer :)

File: src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java (hunk 1 above)
Comment (bereng) [reply src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java]: Yep, leftovers to be removed while I was waiting for one of the original authors to validate the approach...

File: src/java/org/apache/cassandra/tools/nodetool/UpgradeSSTable.java (hunk 2 above)
Comment (bereng) [reply src/java/org/apache/cassandra/tools/nodetool/UpgradeSSTable.java:67]: :shrug: I don't think we care being interrupted or not here. If we do, so be it imo?

File: src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java (hunk 3 above)
Comment (bereng) [reply src/java/org/apache/cassandra/db/compaction/CompactionStrategyManager.java]: Yep they are in their own isolated commit to be dropped annotated with 'DO NOT MERGE'

**Comment by bereng** [comment]
CI all [green](https://app.circleci.com/pipelines/github/bereng/cassandra/1181/workflows/fe2ac859-f6ba-4f1e-b0b1-e6923b16e874) besides `test_change_durable_writes` for `latest` which passes locally. So I think this is a new flaky and I would propose opening a new ticket. Wdyt?

**Pull request closed without merging**